app/src/main/java/org/akvo/flow/api/FlowApi.java

```diff
@@ -16,67 +16,198 @@
         PHONE_NUMBER = StatusUtil.getPhoneNumber(context);
         IMEI = StatusUtil.getImei(context);
     }
-    
-    public List<SurveyedLocale> getSurveyedLocales(long surveyGroup, String timestamp)
-            throws IOException {
-        Context context = FlowApp.getApp();
+
+    public String getServerTime(@NonNull String serverBase) throws IOException {
+        final String url = serverBase + TIME_CHECK_PATH + "?ts=" + System.currentTimeMillis();
+        String response = HttpUtil.httpGet(url);
+        String time = "";
+        if (!TextUtils.isEmpty(response)) {
+            JSONObject json;
+            try {
+                json = new JSONObject(response);
+                time = json.getString("time");
+            } catch (JSONException e1) {
+                Log.e(TAG, "Error fetching time: ", e1);
+            }
+        }
+        return time;
+    }
+
+    /**
+     * Sends the location beacon to the server
+     *
+     * The response is ignored
+     */
+    public void sendLocation(@Nullable String serverBase, @Nullable Double latitude, @Nullable Double longitude,
+                             @Nullable Float accuracy) {
+        if (serverBase != null) {
+            try {
+                String url = serverBase + BEACON_SERVICE_PATH + "&" + getDeviceParams();
+                if (latitude != null && longitude != null && accuracy != null) {
+                    url += LAT + latitude + LON + longitude + ACC + accuracy;
+                }
+                url += OS_VERSION + encodeUrl("Android " + android.os.Build.VERSION.RELEASE);
+                HttpUtil.httpGet(url);
+            } catch (IOException e) {
+                Log.e(TAG, "Could not send location beacon", e);
+            }
+        }
+    }
+
+    /**
+     * Request the notifications GAE has ready for us, like the list of missing files.
+     *
+     * @return String body of the HTTP response
+     *
+     * @throws Exception
+     */
+    @Nullable
+    public JSONObject getDeviceNotification(@NonNull String serverBase, @NonNull String surveyIds) throws Exception {
+        // Send the list of surveys we've got downloaded, getting notified of the deleted ones
+        String url = serverBase + DEVICE_NOTIFICATION_PATH + "?" + getDeviceParams() + surveyIds;
+        String response = HttpUtil.httpGet(url);
+        if (!TextUtils.isEmpty(response)) {
+            return new JSONObject(response);
+        }
+        return null;
+    }
+
+    public List<Survey> getSurveyHeader(String serverBaseUrl, String surveyId) throws IOException {
+        final String url = serverBaseUrl + SURVEY_HEADER_SERVICE_PATH + surveyId + "&" + getDeviceParams();
+        String response = HttpUtil.httpGet(url);
+        if (response != null) {
+            return new SurveyMetaParser().parseList(response, true);
+        }
+        return Collections.emptyList();
+    }
+
+    public List<Survey> getSurveys(String serverBase, List<Survey> surveys) throws IOException {
```

Comment:
I was wondering whether here it's ok to eliminate the second parameter since we will return the list of surveys anyway. i.e we done modify it but rather overwrite the reference.